pkg/lint/rules/template.go

```diff
@@ -235,6 +239,12 @@ func validateYamlContent(err error) error {
 	return nil
 }
 
+// isListResource returns true if the resource is a Kubernetes List type
+// (e.g. ConfigMapList, SecretList).
+func isListResource(obj *k8sYamlStruct) bool {
+	return strings.HasSuffix(obj.Kind, "List")
```

Comment:
I'm wondering if we should be more defensive here. 
For example, treat as a list only if:
- kind == "List", or
- kind ends with "List" AND the top-level object has an items field that is an array.

This would avoid incorrectly skipping `metadata.name` validation for custom resources  that happen to end with "List" but aren't actually list containers. While uncommon, a CRD could define a Kind like "AccessList" that represents a single resource, not a collection.

The items field check should be reliable since all Kubernetes list types 
(ConfigMapList, SecretList, etc.) have this structure, but non-list resources 
typically wouldn't. 🤷

Reply:
Done.